poseidon/poseidon.go

```diff
@@ -177,33 +160,34 @@ func Hash(arr []*big.Int) (*big.Int, error) {
 		toHash[j] = r
 		j++
 		for ; j < T; j++ {
-			toHash[j] = constants.Zero
+			toHash[j] = Zero()
 		}
 
 		ph, err := PoseidonHash(toHash)
 		if err != nil {
 			return nil, err
 		}
-		modQ(r.Add(r, ph))
+		r.Add(r, ph)
 	}
 
 	return r, nil
 }
 
 // HashBytes hashes a msg byte slice by blocks of 31 bytes encoded as
 // little-endian
-func HashBytes(b []byte) (*big.Int, error) {
+func HashBytes(b []byte) (*ff.Element, error) {
```

Comment:
Same as before: maybe keep `*big.Int` for the public API

Reply:
Done (same as before comment)